atomic_reactor/plugins/exit_koji_import.py

```diff
@@ -0,0 +1,257 @@
+    be used (from the kernel keyring) unless both koji_keytab and
+    koji_principal are specified. The koji_keytab parameter is a
+    keytab name like 'type:name', and so can be used to specify a key
+    in a Kubernetes secret by specifying 'FILE:/path/to/key'.
+
+    Runs as an exit plugin in order to capture logs from all other
+    plugins.
+    """
+
+    key = PLUGIN_KOJI_IMPORT_PLUGIN_KEY
+    is_allowed_to_fail = False
+
+    def __init__(self, tasker, workflow, kojihub, url,
+                 verify_ssl=True, use_auth=True,
+                 koji_ssl_certs=None, koji_proxy_user=None,
+                 koji_principal=None, koji_keytab=None,
+                 target=None, poll_interval=5):
+        """
+        constructor
+
+        :param tasker: DockerTasker instance
+        :param workflow: DockerBuildWorkflow instance
+        :param kojihub: string, koji hub (xmlrpc)
+        :param url: string, URL for OSv3 instance
+        :param verify_ssl: bool, verify OSv3 SSL certificate?
+        :param use_auth: bool, initiate authentication with OSv3?
+        :param koji_ssl_certs: str, path to 'cert', 'ca', 'serverca'
+        :param koji_proxy_user: str, user to log in as (requires hub config)
+        :param koji_principal: str, Kerberos principal (must specify keytab)
+        :param koji_keytab: str, keytab name (must specify principal)
+        :param target: str, koji target
+        :param poll_interval: int, seconds between Koji task status requests
+        """
+        super(KojiImportPlugin, self).__init__(tasker, workflow)
+
+        self.kojihub = kojihub
+        self.koji_ssl_certs = koji_ssl_certs
+        self.koji_proxy_user = koji_proxy_user
+
+        self.koji_principal = koji_principal
+        self.koji_keytab = koji_keytab
+
+        self.target = target
+        self.poll_interval = poll_interval
+
+        self.namespace = get_build_json().get('metadata', {}).get('namespace', None)
+        osbs_conf = Configuration(conf_file=None, openshift_uri=url,
+                                  use_auth=use_auth, verify_ssl=verify_ssl,
+                                  namespace=self.namespace)
+        self.osbs = OSBS(osbs_conf, osbs_conf)
+        self.build_id = None
+
+    def get_output(self, worker_metadatas):
+        """
+        Build the output entry of the metadata.
+
+        :return: list, containing dicts of partial metadata
+        """
+        outputs = []
+        for platform in worker_metadatas:
+            for instance in worker_metadatas[platform]['output']:
+                instance['buildroot_id'] = platform + '-' + repr(instance['buildroot_id'])
+                outputs.append(instance)
+
+        return outputs
+
+    def get_buildroot(self, worker_metadatas):
+        """
+        Build the buildroot entry of the metadata.
+
+        :return: list, containing dicts of partial metadata
+        """
+        buildroots = []
+
+        # force ordering to work around python2 to python3 bug
```

Comment:
Not really a bug. A `dict` is not guaranteed to be ordered. The order will be different when the hashing algorithm changes. For retaining order, use `ordereddict`.

In this case, if all you want is to use a sorted list of platforms:
```python
for platform in sorted(worker_metadatas.keys()):
   for instance in worker_metadatas[platform]['buildroots']:
      ...
```